test/setup/test_provisioner.py

```diff
@@ -56,6 +65,25 @@ def test_config_apply_bootstrap(self):
 
         self.assertEqual(rc, 0)
 
+    def test_add_num_keys(self):
+        """Test add_num_keys interface."""
+        data = {
+                'a' : ['1', '2', '3'],
+                'b' : '4',
+                'c' : [{'5' : ['6', '7']}, '8']
+        }
+        test_index = 'test_index'
+        working_directory = os.path.realpath(sys.argv[0])
+        config_path = os.path.join(working_directory,"/sample_config.yaml")
+        with open(config_path, 'w+') as config:
+            config.write(yaml.dump(data))
+        config_url = f"yaml://{config_path}"
+        cortx_conf = MappedConf(config_url)
+        CortxProvisioner._add_num_keys(cortx_conf)
+        Conf.load(test_index, config_url)
+        self.assertEqual(3, Conf.get(test_index, 'num_a'))
+        self.assertEqual(2, Conf.get(test_index, 'c[0]>num_5'))
+        delete_file(config_path)
```

Comment:
Also add a test where
- load config
- add num keys (dont save)
- load config into another index
- num_keys must not be present

Reply:
done